src/de/mod.rs

```diff
@@ -870,6 +867,46 @@ mod tests {
         );
     }
 
+    #[test]
+    fn struct_with_array_field() {
+        #[derive(Debug, Deserialize, PartialEq)]
+        struct Test {
+            status: bool,
+            point: [u32; 3],
+        }
+
+        let (result, _consumed): (Test, _) =
+            crate::from_str(r#"{ "status": true, "point": [1,2,3] }"#).expect("it deserializes");
+
+        assert_eq!(
+            result,
+            Test {
+                status: true,
+                point: [1_u32, 2, 3]
+            }
+        );
+    }
+
+    #[test]
+    fn struct_with_tuple_field() {
+        #[derive(Debug, Deserialize, PartialEq)]
+        struct Test {
+            status: bool,
+            point: (u32, u32, u32),
+        }
+
+        let (result, _consumed): (Test, _) =
+            crate::from_str(r#"{ "status": true, "point": [1,2,3] }"#).expect("it deserializes");
```

Comment:
Shouldn't the message in expect() be an error message displayed to the user? E.g. deserialization unsuccessful?

Reply:
changed test to match the style of the other tests